core/src/apps/bitcoin/common.py

```diff
@@ -47,11 +57,13 @@
 SEGWIT_INPUT_SCRIPT_TYPES = (
     InputScriptType.SPENDP2SHWITNESS,
     InputScriptType.SPENDWITNESS,
+    InputScriptType.SPENDTAPROOT,
```

Comment:
Correct. Since in case (b) the code is unreachable for taproot, there are no implications. When we implement ownership proofs for taproot the code will be reachable and it will have undergone some changes. But, for example the principle that non-taproot inputs have `b"\x00"` as the witness will remain.

Reply:
It would just wrap the provided signature into an invalid witness script, i.e. the returned script wouldn't work.